students/ericstreit/lessons/lesson09/assignment/database.py

```diff
@@ -0,0 +1,238 @@
+"""
+#Lesson10
+Metaprogramming Timing
+Special Thanks to:
+https://www.blog.pythonlibrary.org/2016/05/24/python-101-an-intro-to-benchmarking-your-code/
+"""
+
+#!/usr/bin/env python3
+#import stuff here
+from pymongo import MongoClient
+import logging
+import csv
+import time
+
+
+#global variables here
+logging.basicConfig(level=logging.ERROR)
+#define function and classes
+
+def timerfunc(func):
+    """ a timer decorator """
+
+    def function_timer(*args, **kwargs):
+        """ nested function for timing functions """
+
+        start = time.time()
+        value = func(*args, **kwargs)
+        end = time.time()
+        runtime = end - start
+        print(f' The runtime for {func.__name__} took {runtime} seconds to complete')
+        return value
+    return function_timer
+
+class MongoDBConnection():
+    """MongoDB Connection"""
+
+    def __init__(self, host='127.0.0.1', port=27017):
+        """ be sure to use the ip address not name for local windows"""
+        self.host = host
+        self.port = port
+        self.connection = None
+
+    def __enter__(self):
+        self.connection = MongoClient(self.host, self.port)
+        print(f"Establishing a connection to {self.host} on port {self.port}")
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        print(f"closing the connection to {self.host}!")
+        self.connection.close()
```

Comment:
Wait, I'm confused. Is this 9 or 10? Is adding print statements your improvement? Probably not what they had in mind.